lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -2465,33 +2766,34 @@ struct ServerDriveQueryDirectoryRequest {
 
 impl ServerDriveQueryDirectoryRequest {
     fn decode(device_io_request: DeviceIoRequest, payload: &mut Payload) -> RdpResult<Self> {
-        let fs_information_class_lvl =
-            FsInformationClassLevel::from_u32(payload.read_u32::<LittleEndian>()?)
-                .ok_or_else(|| invalid_data_error("failed to read FsInformationClassLevel"))?;
-        if fs_information_class_lvl != FsInformationClassLevel::FileDirectoryInformation
-            && fs_information_class_lvl != FsInformationClassLevel::FileFullDirectoryInformation
-            && fs_information_class_lvl != FsInformationClassLevel::FileBothDirectoryInformation
-            && fs_information_class_lvl != FsInformationClassLevel::FileNamesInformation
-        {
+        let file_info_class_lvl =
+            FileInformationClassLevel::from_u32(payload.read_u32::<LittleEndian>()?)
+                .ok_or_else(|| invalid_data_error("failed to read FileInformationClassLevel"))?;
+
+        // These are all the FsInformationClass's supported for this message by FreeRDP
+        // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L794
+        let valid_levels = vec![
```

Comment:
```suggestion
        let valid_levels = &[
```
An array literal works without allocating

Reply:
nit: this should be a static array, not declared with let

Reply:
even better